### ISSUE TITLE
Smileys not sent to skype

### ISSUE THREAD

**Issue by davidhealey** [EionRobb/skype4pidgin#666]
Sending a smiley such as :D shows up fine in Pidgin but it isn't forwarded to Skype correctly. In Skype it just shows up as the two characters.

**Comment by alexolog** [comment]
This is not a Skype plugin issue, as it affects all the protocols.  Pidgin does not seem to translate outgoing text while the native clients do -- each one according to it's proprietary rules.

Now, if you want to simulate the behaviour of the native Skype client, you can do it with the " Text Replacement" plug-in by replacing its dictionary file with one that "corrects" those strings to the appropriate Emoji.  However, keep in mind that Pidgin has issues with Emoji on Windows, so you may want to use a smiley theme as well.

I have made one for my personal use, and shared it [on Reddit](https://www.reddit.com/r/pidgin/comments/ievaq1/unicode_smiley_theme_based_on_joypixels_emojis/).  It is Unicode Emoji based, so it does not have Skype-specific ones, but you can use it as a starting point.

You can find the full list of Skype emoticons here:
http://www.skaip.org/skype-emoticons

**Comment by EionRobb** [comment]
It kind of is the plugin, Skype servers expect smileys to be wrapped in `<ss>` xml tags, which Pidgin doesn't send.

The plugin could potentially try to do a regex replace to turn `;)` into `<ss type="wink">;)</ss>` but I'd worry about getting the regex wrong and embedding it in the middle of a url or code-block or something

**Comment by davidhealey** [comment]
> It kind of is the plugin, Skype servers expect smileys to be wrapped in `<ss>` xml tags, which Pidgin doesn't send.
> 
> The plugin could potentially try to do a regex replace to turn `;)` into `<ss type="wink">;)</ss>` but I'd worry about getting the regex wrong and embedding it in the middle of a url or code-block or something

So if I use the text replacement plugin as suggested by alex, combined with the ss tags, it should display okay in skype and look slightly weird in Pidgin (which I'm fine with)?

**Comment by davidhealey** [comment]
Ah I just tested it and it just shows the entire string in Skype too

**Comment by alexolog** [comment]
I have the text replacement plugin convert `:)` into the appropriate Unicode emoji.
When I type it in the input box, it gets converted, and when I send, the recipient sees this:
![image](https://user-images.githubusercontent.com/2112513/94374840-2268d480-00dd-11eb-88fd-fbe910a6dab5.png)

Obviously it will not work with Skype emoticons that have no Unicode emoji analogs

**Comment by EionRobb** [comment]
@davidhealey yeah, sorry, the text replacement plugin doesn't understand xml/html it's replacing the plain-text - you could try copy pasting some HTML with the `<ss>` tag, but that's fairly high-effort for little gain :(

**Comment by alexolog** [comment]
(edited comment above)

**Comment by davidhealey** [comment]
> I have the text replacement plugin convert `:)` into the Unicode emoji `🙂`:
> 
> ```
> COMPLETE 0
> CASE 1
> BAD :)
> GOOD 🙂
> ```
> 
> When I type it in the input box, it gets converted, and when I send, the recipient sees this:
> ![image](https://user-images.githubusercontent.com/2112513/94374840-2268d480-00dd-11eb-88fd-fbe910a6dab5.png)
> 
> Obviously it will not work with Skype emoticons that have no Unicode emoji analogs

Skype doesn't recognise 🙂: it just shows up as the characters

**Comment by alexolog** [comment]
Are you sure it got converted?  The text replacement plugin is supposed to replace the text in the input window, as you are typing it.  When I type `:` followed by `)`, the `:` immediately disappears to be replaced with a Unicode "tofu".  The actual characters I typed do not get sent.

**Comment by davidhealey** [comment]
This is what I see in Pidgin 
![image](https://user-images.githubusercontent.com/14021202/94375513-11838780-010c-11eb-8b78-41b355d6bca7.png)

And this is the result I see in Skype
![image](https://user-images.githubusercontent.com/14021202/94375519-2233fd80-010c-11eb-9eff-8bd0ecb590f6.png)

**Comment by alexolog** [comment]
Hmmm...  I see the smiley in the Skype web interface

**Comment by alexolog** [comment]
Could be that the font you are using in Skype does not support Unicode Emoji?

**Comment by davidhealey** [comment]
Oh that colon you're seeing is actually at the end of my name in the chat. `Name: []`

I'm using web skype too so I must be doing something wrong in Pidgin. Although I tried pasting it directly into Skype too

**Comment by bodqhrohro** [comment]
There is a DeWYSIWYGification plugin from purple-plugin-pack to send raw tags; I just checked it with skype4pidgin, works well.

Maybe skype4pidgin should add a toolbar button with Skype-specific smiles (do prpls have an access to it or this should be implemented as a separate plugin?)

**Comment by alexolog** [comment]
Wouldn't it interfere with other things, like sending URLs?
